rspace/src/main/scala/coop/rchain/rspace/LMDBStorage.scala

```diff
@@ -0,0 +1,94 @@
+package coop.rchain.rspace
+
+import java.nio.ByteBuffer
+
+import org.lmdbjava.{Dbi, Env, Txn}
+import scodec.Codec
+import coop.rchain.shared.ByteVectorOps._
+import coop.rchain.shared.AttemptOps._
+import scodec.bits.BitVector
+
+trait LMDBStorage {
+  private[rspace] def env: Env[ByteBuffer]
+
+  private[rspace] def createTxnRead(): Txn[ByteBuffer] = env.txnRead
+
+  private[rspace] def createTxnWrite(): Txn[ByteBuffer] = env.txnWrite
+
+  private[rspace] def withTxn[R](txn: Txn[ByteBuffer])(f: Txn[ByteBuffer] => R): R =
+    try {
+      val ret: R = f(txn)
+      txn.commit()
+      ret
+    } catch {
+      case ex: Throwable =>
+        txn.abort()
+        throw ex
+    } finally {
+      txn.close()
+    }
+
+  /** These methods make an assumption that is true in the current execution,
```

Comment:
and what is that assumption?

Reply:
also this comment only really scopes to the first three methods (namely `put`, `delete`, and `get` with the `Blake2b256Hash` parameters), right ?

Reply:
I'll reword it and make it point at the specialised methods